src/coord/src/coord.rs

```diff
@@ -204,8 +204,11 @@ pub struct LoggingConfig {
 }
 
 /// Configures a coordinator.
-pub struct Config<'a> {
-    pub dataflow_client: dataflow::Client,
+pub struct Config<'a, C>
+where
+    C: dataflow::Client,
+{
+    pub dataflow_client: C,
```

Comment:
Is it possible that this could be `dataflow_client: dyn Box<dataflow::Client>` to stem any flow of complexity? Happy to have it this way ofc, but just putting that out there. The API would be fine with a virtual call on each method invocation, I think.

Reply:
Yeah, agree! I went back and forth on this. Happy to have it move around.